cmake/HPX_SetupTarget.cmake

```diff
@@ -224,6 +224,13 @@ function(hpx_setup_target target)
     target_link_libraries(${target} ${__tll_private} hpx_internal_flags)
   endif()
 
+  # if we build tests using an installed version of HPX we need to add the local
+  # root as an include path as tests refer to headers that are not normally
+  # installed
+  if(HPX_WITH_INSTALLED_VERSION)
+    target_include_directories(${target} ${__tll_private} ${PROJECT_SOURCE_DIR})
+  endif()
+
```

Comment:
Could this not be done only for the targets that need it? Or are there so many of them that it makes sense to do it like this instead?

Reply:
Fair point. I guess I was just too lazy to figure out what tests need it. It is quite a list of them, however.

Reply:
From a quick search it looks like `tests/unit/agas`, `tests/unit/component`, `tests/regressions/actions` might be enough. This was searching for `#include <tests/`. There may be more though, in which case adding it for every target is fair. Relative `#include "..."`s would also be an option.

Reply:
I changed everything to relative #include paths.